### PR TITLE
Add a shared id for matching one-time-scans together

### PR DESCRIPTION
**Pull request by FestiveKyle**
As described

### CODE REVIEW

File: api-js/src/email-scan/objects/dkim-sub.js [hunk 1]
```diff
@@ -8,6 +8,11 @@ export const dkimSubType = new GraphQLObjectType({
   description:
     'DKIM gql object containing the fields for the `dkimScanData` subscription.',
   fields: () => ({
+    sharedId: {
+      // TODO type: ID_TYPE,
```
Comment (nsdeschenes) [inline api-js/src/email-scan/objects/dkim-sub.js:12]: `type: GraphQLID`

File: api-js/src/email-scan/objects/dkim-sub.js [hunk 2]
```diff
@@ -8,6 +8,11 @@ export const dkimSubType = new GraphQLObjectType({
...
+      description: `The shared id to match scans together.`,
+      resolve: async ({ sharedId }) => sharedId,
```
Comment (nsdeschenes) [inline api-js/src/email-scan/objects/dkim-sub.js:14]: No async needed

File: api-js/src/email-scan/objects/dmarc-sub.js [hunk 3]
```diff
@@ -14,6 +14,11 @@ export const dmarcSubType = new GraphQLObjectType({
   description:
     'DMARC gql object containing the fields for the `dkimScanData` subscription.',
   fields: () => ({
+    sharedId: {
+      // TODO type: ID_TYPE,
```
Comment (nsdeschenes) [inline api-js/src/email-scan/objects/dmarc-sub.js:18]: :point_up:

File: api-js/src/email-scan/objects/dmarc-sub.js [hunk 4]
```diff
@@ -14,6 +14,11 @@ export const dmarcSubType = new GraphQLObjectType({
...
+      description: `The shared id to match scans together.`,
+      resolve: async ({ sharedId }) => sharedId,
```
Comment (nsdeschenes) [inline api-js/src/email-scan/objects/dmarc-sub.js:20]: :point_up:

File: api-js/src/email-scan/objects/spf-sub.js [hunk 5]
```diff
@@ -14,6 +14,11 @@ export const spfSubType = new GraphQLObjectType({
   description:
     'SPF gql object containing the fields for the `dkimScanData` subscription.',
   fields: () => ({
+    sharedId: {
+      // TODO type: ID_TYPE,
```
Comment (nsdeschenes) [inline api-js/src/email-scan/objects/spf-sub.js:18]: :point_up:

File: api-js/src/email-scan/objects/spf-sub.js [hunk 6]
```diff
@@ -14,6 +14,11 @@ export const spfSubType = new GraphQLObjectType({
...
+      description: `The shared id to match scans together.`,
+      resolve: async ({ sharedId }) => sharedId,
```
Comment (nsdeschenes) [inline api-js/src/email-scan/objects/spf-sub.js:20]: :point_up:

File: api-js/src/web-scan/objects/https-sub.js [hunk 7]
```diff
@@ -9,6 +9,11 @@ export const httpsSubType = new GraphQLObjectType({
   description:
     'HTTPS gql object containing the fields for the `dkimScanData` subscription.',
   fields: () => ({
+    sharedId: {
+      // TODO type: ID_TYPE,
```
Comment (nsdeschenes) [inline api-js/src/web-scan/objects/https-sub.js:13]: :point_up:

File: api-js/src/web-scan/objects/https-sub.js [hunk 8]
```diff
@@ -9,6 +9,11 @@ export const httpsSubType = new GraphQLObjectType({
...
+      description: `The shared id to match scans together.`,
+      resolve: async ({ sharedId }) => sharedId,
```
Comment (nsdeschenes) [inline api-js/src/web-scan/objects/https-sub.js:15]: :point_up:

File: api-js/src/web-scan/objects/ssl-sub.js [hunk 9]
```diff
@@ -14,6 +14,11 @@ export const sslSubType = new GraphQLObjectType({
   description:
     'SSL gql object containing the fields for the `dkimScanData` subscription.',
   fields: () => ({
+    sharedId: {
+      // TODO type: ID_TYPE,
```
Comment (nsdeschenes) [inline api-js/src/web-scan/objects/ssl-sub.js:18]: :point_up:

File: api-js/src/web-scan/objects/ssl-sub.js [hunk 10]
```diff
@@ -14,6 +14,11 @@ export const sslSubType = new GraphQLObjectType({
...
+      description: `The shared id to match scans together.`,
+      resolve: async ({ sharedId }) => sharedId,
```
Comment (nsdeschenes) [inline api-js/src/web-scan/objects/ssl-sub.js:20]: :point_up:

**Review by nsdeschenes — CHANGES_REQUESTED** [review]
Need the tests for keeping that sweet sweet coverage up!

**Review by Ethanljf — APPROVED** [review]
🔥

**Review by ThomasNickerson — APPROVED** [review]

**Review by nsdeschenes — APPROVED** [review]

**Pull request merged**